lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -98,18 +101,53 @@ def deploy_failing_to_progress?
       Time.parse(@progress_condition["lastUpdateTime"]).to_i >= (@deploy_started_at - 5.seconds).to_i
     end
 
-    def find_latest_rs(deployment_data)
-      label_string = deployment_data["spec"]["selector"]["matchLabels"].map { |k, v| "#{k}=#{v}" }.join(",")
+    def all_rs_data(matchLabels)
+      label_string = matchLabels.map { |k, v| "#{k}=#{v}" }.join(",")
       raw_json, _err, st = kubectl.run("get", "replicasets", "--output=json", "--selector=#{label_string}")
-      return unless st.success?
+      return {} unless st.success?
+
+      JSON.parse(raw_json)["items"]
+    end
+
+    def running_rs
+      all_rs_data(@deployment_data["spec"]["selector"]["matchLabels"]).select do |rs|
+        rs["status"]["replicas"].to_i > 0
+      end
+    end
+
+    def min_unavailable_rollout
+      @definition.dig('metadata', 'annotations', 'kubernetes-deploy.shopify.io/min-unavailable-rollout')
+    end
 
-      all_rs_data = JSON.parse(raw_json)["items"]
-      current_revision = deployment_data["metadata"]["annotations"]["deployment.kubernetes.io/revision"]
+    def minimum_unavailable_replicas_to_succeeded
+      desired = @desired_replicas
 
-      latest_rs_data = all_rs_data.find do |rs|
-        rs["metadata"]["ownerReferences"].any? { |ref| ref["uid"] == deployment_data["metadata"]["uid"] } &&
+      if min_unavailable_rollout =~ /%/
+        desired *= (100 - min_unavailable_rollout.to_i) / 100.0
+      else
+        desired = min_unavailable_rollout.to_i
+      end
+
+      desired.to_i
+    end
+
+    def partial_deployment_success
+      minimum_needed = minimum_unavailable_replicas_to_succeeded
+
+      running_rs.size <= 2 &&
```

Comment:
I can understand the motivation for this check, but couldn't it make the new feature ineffective for deployments with long-running jobs (like core has) that take a long time to terminate? We actually have a test asserting that terminating pods from older revisions do not impede deploy success. I'm not certain what that state looks like from a RS standpoint, but I'd expect it to be `desired = 0` and `replicas > 0`.